tests/src/JIT/HardwareIntrinsics/X86/General/VectorArray.cs

```diff
@@ -0,0 +1,615 @@
+                }
+            }
+            else if (typeof(T) == typeof(sbyte))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<sbyte, T>(Sse2.Add(Sse.StaticCast<T, sbyte>(pos[i]), Sse.StaticCast<T, sbyte>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(short))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<short, T>(Sse2.Add(Sse.StaticCast<T, short>(pos[i]), Sse.StaticCast<T, short>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(ushort))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<ushort, T>(Sse2.Add(Sse.StaticCast<T, ushort>(pos[i]), Sse.StaticCast<T, ushort>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(int))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<int, T>(Sse2.Add(Sse.StaticCast<T, int>(pos[i]), Sse.StaticCast<T, int>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(uint))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<uint, T>(Sse2.Add(Sse.StaticCast<T, uint>(pos[i]), Sse.StaticCast<T, uint>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(long))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<long, T>(Sse2.Add(Sse.StaticCast<T, long>(pos[i]), Sse.StaticCast<T, long>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(ulong))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<ulong, T>(Sse2.Add(Sse.StaticCast<T, ulong>(pos[i]), Sse.StaticCast<T, ulong>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(float))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<float, T>(Sse.Add(Sse.StaticCast<T, float>(pos[i]), Sse.StaticCast<T, float>(delta)));
+                }
+            }
+            else if (typeof(T) == typeof(double))
+            {
+                for (int i = 0; i < pos.Length; ++i)
+                {
+                    pos[i] = Sse.StaticCast<double, T>(Sse2.Add(Sse.StaticCast<T, double>(pos[i]), Sse.StaticCast<T, double>(delta)));
+                }
+            }
+            else
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        static public unsafe int Vector128Array(T deltaValue)
```

Comment:
I see that this applies to the existing test that you based this on, but `deltaValue` doesn't seem to be used. I presume that it was intended to be the value used to initialize the `delta` vector, but I'm not sure.